pkgs/coverage/lib/src/collect.dart

```diff
@@ -262,7 +264,17 @@ Future<List<Map<String, dynamic>>> _processSourceReport(
     return scripts[scriptRef];
   }
 
-  HitMap getHitMap(Uri scriptUri) => hitMaps.putIfAbsent(scriptUri, HitMap.new);
+  HitMap getHitMap(Uri scriptUri) => hitMaps.putIfAbsent(scriptUri, () {
+        final hits = HitMap();
```

Comment:
Consider adding a new constructor for `HitMap` that performs this initialization.